Toasts.Forms.Plugin.Droid/NotificationBuilder.cs

```diff
@@ -1,95 +1,145 @@
+using Android.Graphics;
+using Android.App;
+using Android.Content;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Xml.Serialization;
+using Plugin.Toasts.Interfaces;
+
 namespace Plugin.Toasts
 {
-    using Android.App;
-    using Android.Content;
-    using System;
-    using System.Collections.Generic;
-    using System.IO;
-    using System.Threading;
-    using System.Xml.Serialization;
-
-    internal class NotificationBuilder
+    class NotificationBuilder
     {
+        public static string PackageName { get; set; }
+
         public static IDictionary<string, ManualResetEvent> ResetEvent = new Dictionary<string, ManualResetEvent>();
         public static IDictionary<string, NotificationResult> EventResult = new Dictionary<string, NotificationResult>();
+        public static List<string> Channels = new List<string>();
 
         public const string NotificationId = "NOTIFICATION_ID";
         public const string DismissedClickIntent = "android.intent.action.DISMISSED";
         public const string OnClickIntent = "android.intent.action.CLICK";
+        public const string NotificationForceOpenApp = "NOTIFICATION_FORCEOPEN";
+
+        public const string DefaultChannelName = "default";
+
+        public const int StartId = 123;
 
-        private int _count = 0;
-        private object _lock = new object();
-        private static NotificationReceiver _receiver;
-        private IPlatformOptions _androidOptions;
+        int _count = 0;
+        object _lock = new object();
+        static NotificationReceiver _receiver;
+        IPlatformOptions _androidOptions;
 
-        public void Init(Activity activity, IPlatformOptions androidOptions)
+        public void Init(IPlatformOptions androidOptions)
         {
+            PackageName = Application.Context.PackageName;
+
             IntentFilter filter = new IntentFilter();
             filter.AddAction(DismissedClickIntent);
             filter.AddAction(OnClickIntent);
 
             _receiver = new NotificationReceiver();
 
-            activity.RegisterReceiver(_receiver, filter);
+            Application.Context.RegisterReceiver(_receiver, filter);
 
             _androidOptions = androidOptions;
         }
 
-        public IList<INotification> GetDeliveredNotifications(Activity activity)
+        public static string GetOrCreateChannel(IAndroidChannelOptions channelOptions)
```

Comment:
Why static?

Reply:
That is static as the AlarmHandler uses it as well to  create the channel for scheduled notifications.